core/src/main/java/org/infinispan/configuration/parsing/Parser.java

```diff
@@ -2779,14 +2779,25 @@ private void parseKeyTransformer(XMLExtendedStreamReader reader, ConfigurationBu
 
    private void parseIndexedEntities(XMLExtendedStreamReader reader, ConfigurationBuilderHolder holder, ConfigurationBuilder builder) throws XMLStreamException {
       ParseUtils.requireNoAttributes(reader);
+      boolean isProtobufStorage = builder.memory().encoding().value().isProtobufStorage();
```

Comment:
Unfortunately  ```builder.memory().encoding().value().isProtobufStorage()``` only gives the information that "protobuf was explicitly configured in the cache", and not if the storage turns out to be protobuf or not. Storage information is obtained only at runtime
`advancedCache.getValueDataConversion().getStorageMediaType()`

I think the reasons why it "works" here in this PR is because the exception is ignored

Reply:
And don't we want to know that "protobuf was explicitly configured in the cache" ?

Reply:
And we need that to know at config time not at runtime.

Reply:
atm there is no way of knowing at config time.

Reply:
we absolutely need that in 12, at least, otherwise the whole pre-declared stuff for HS 6 will not really exist.

Reply:
yes, it shall be available at config time